File: testers/avro/scala/generated-and-checked-in/com/example/service/UserService.scala
package com.example.service



/** User management service protocol */
trait UserService {
  /** Get a user by their ID */
  def getUser(userId: String): Result[User, UserNotFoundError]

  /** Create a new user */
  def createUser(
    email: String,
    name: String
  ): Result[User, ValidationError]

  /** Delete a user */
  def deleteUser(userId: String): Result[Unit, UserNotFoundError]

  /** Send a notification to a user (fire-and-forget) */
  def notifyUser(
    userId: String,
    message: String
  ): Unit
}

File: testers/avro/kotlin-quarkus-mutiny/generated-and-checked-in/com/example/service/UserService.kt
package com.example.service

import io.smallrye.mutiny.Uni
import java.lang.Void

/** User management service protocol */
interface UserService {
  /** Create a new user */
  abstract fun createUser(
    email: kotlin.String,
    name: kotlin.String
  ): Uni<Result<User, ValidationError>>

  /** Delete a user */
  abstract fun deleteUser(userId: kotlin.String): Uni<Result<Unit, UserNotFoundError>>

  /** Get a user by their ID */
  abstract fun getUser(userId: kotlin.String): Uni<Result<User, UserNotFoundError>>

  /** Send a notification to a user (fire-and-forget) */
  abstract fun notifyUser(
    userId: kotlin.String,
    message: kotlin.String
  ): Uni<Void>
}